botocore/client.py

```diff
@@ -117,16 +114,12 @@ def create_client(
         )
         service_name = first_non_none_response(responses, default=service_name)
         service_model = self._load_service_model(service_name, api_version)
-
-        if (
-            service_name in ENDPOINT_RESOLUTION_V2_SERVICES
-            or FORCE_ENDPOINT_RESOLUTION_V2
-        ):
+        try:
             endpoints_ruleset_data = self._load_service_endpoints_ruleset(
                 service_name, api_version
             )
             partition_data = self._loader.load_data('partitions')
-        else:
+        except UnknownServiceError:
             endpoints_ruleset_data = None
             partition_data = None
```

Comment:
Can we do a quick info log here stating something like:

> Unknown service {service_name} detected, attempting to fallback to legacy endpoint routing.

Reply:
Done in 1cafc36.